equinox/serialisation.py

```diff
@@ -202,17 +237,16 @@ def tree_deserialise_leaves(
     **Arguments:**
 
     - `path`: The file location to load values from.
-    - `like`: A PyTree of the same structure, and with leaves of the same type, as the
-        PyTree being loaded. Those leaves which are loaded will replace the
-        corresponding leaves of `like`.
+    - `like`: A PyTree of same or overlapping depth-first structure of the saved PyTree.
+        The matching leaves are required to be of the same type.
+        Those leaves which are loaded will replace the corresponding leaves of `like`.
```

Comment:
I'd like to revert the change here. I think we should continue to specify that it must be exactly the same structure.